proposals/0000-borrow-operator.md

```diff
@@ -0,0 +1,599 @@
+from that document, our understanding of some of these concepts has
+changed, and the terms we've used for them has changed as well.
+I'll start by restating the definitions from the manifesto using
+modern terminology.
+
+An *high-level value* is our human understanding of the information
+content of the result of an expression. The expression `[1,2,3]`
+creates an array of three integers. After the following code runs,
+the variables `a` and `b` hold the same high-level values because
+they currently contain the same information.
+
+```swift
+var a = [1, 2, 3]
+var b = a
+```
+
+The variable `c` has this same high-level value, even though the
+value is constructed completely independently:
+
+```swift
+var c = [1]
+c += [2, 3]
+```
+
+A *low-level value* is one step down in abstraction level. `a`, `b`, and `c`
+all store different low-level values. If we carefully read the standard
+library source code, we can see that a low-level value of type `Array<Int>`
+is an owned reference to an array buffer object. The low-level value in `b`
+was produced by copying the low-level value in `a`. Copying an owned object reference produces a new owned reference to the same object, so these two
+values now refer to the same buffer. However, they are still different
+low-level values because they each have independent ownership of a
+reference to that object.
+
+That ownership is one of the most important components of these low-level
+values, and it is intangible --- it cannot be directly observed. If you
+inspected this program's memory in a debugger, you might be able to see
+that the reference count of the array buffer object referenced by `a` and
+`b` is 2, but you would not see that the variables have responsibility for
+a single reference count apiece. An unowned reference to that object might
+look the same in memory, but it would not have the same responsibility.[^1]
+
+[^1]: Swift's `unowned` references are memory-safe, so in fact they do still
+have a responsibility to manage the object, it's just a *different*
+responsibility from an owned reference. `unowned(unsafe)` references are
+irresponsible, which is why they're unsafe.
+
+If we went another step down in abstraction level, we would need to get into
+*representations* of values and how they are laid out in memory. But in this
+proposal, we are concerned with ownership, which mostly operates at the level
+of low-level values. So we do not need to talk about representations, and
+we will just talk about low-level values.
+
+For the rest of this proposal, the word *value* by itself should be understood
+to mean this low-level concept in which copying a value produces a new value.
+
+Every use of a value in Swift has some interaction with ownership.
+Non-mutating uses of values are always either:
+
+- *borrows*, which just read the low-level information in the value, or
+- *copies*, which create a new, independent copy of that low-level
+  information according to the rules of the type.
+
```

Comment:
I wonder if there should be a footnote here that points out that "borrow" does not necessarily mean "pointer to."  For small items, it's more efficient to "bitwise borrow" by (temporarily) copying the contents.